scripts/plugins/anchor-markdown-headings.js

```diff
@@ -12,12 +12,19 @@
  * `<!---->` to quote your English anchor name inside, with your
  * own title beside it.
  */
-module.exports = function anchorMarkdownHeadings (text, level, raw, slugger) {
-  // Check whether we've got the comment matches
-  // <!--comment-->, <!--comment --> and even <!-- comment-->
-  // (20 hex = 32 dec = space character)
-  const ANCHOR_COMMENTREG = /<!--\x20?([\w|-]+)\x20?-->/gi
 
+// Check whether we've got the comment matches
+// <!--comment-->, <!--comment --> and even <!-- comment-->
+// (20 hex = 32 dec = space character)
+// Only need to find one that matches the Regex, we cannot use
+// 'g' here because it will continue searching for the rest string,
+// and you have to reset the lastIndex, which isn't a necessary for
+// our current situation.
+// For more, you can see:
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/lastIndex#Description
+const ANCHOR_COMMENTREG = /<!--\x20?([\w\x20-]+)\x20?-->/i
```

Comment:
I still wonder why this can't be

```js
const ANCHOR_COMMENTREG = /<!-- ?([\w -]+) ?-->/
```

Reply:
This is also OK, but I tend to use \x20 or \s, something like this as a symbol.

Reply:
I also removed `i` since we don't care about the case here. `\s` is fine too, but it also allows newlines apart from spaces and tabs.

Anyway just thinking out loud.

Reply:
Ah I see, \s is more strict than what I meant.

Reply:
Which we don't want in our case since we don't use newlines in the comment. But removing `i` seems fine.

Reply:
OK, I'll submit again. Thanks for your patience